Marlin/src/module/stepper.h

```diff
@@ -331,34 +331,36 @@ constexpr ena_mask_t enable_overlap[] = {
 #if HAS_SHAPING
 
   // These constexpr are used to calculate the shaping queue buffer sizes
-  constexpr xyze_float_t max_feedrate = DEFAULT_MAX_FEEDRATE;
-  constexpr xyze_float_t steps_per_unit = DEFAULT_AXIS_STEPS_PER_UNIT;
+  struct DistinctAxes{ float x, y, z, i, j, k, u, v, w, e0, e1, e2, e3, e4, e5, e6, e7; };
+  constexpr DistinctAxes max_feedrate = DEFAULT_MAX_FEEDRATE;
+  constexpr DistinctAxes steps_per_unit = DEFAULT_AXIS_STEPS_PER_UNIT;
+  constexpr float _max_feedrate[] = DEFAULT_MAX_FEEDRATE;
+  constexpr float _steps_per_unit[] = DEFAULT_AXIS_STEPS_PER_UNIT;
```

Comment:
These `DistinctAxes` changes are definitely not going to work. Yes, it compiles, because C++ inserts 0 for missing initializers. But these configuration settings must match the actual axes that are configured.

Reply:
`xyze_float_t` does not work when there is more than one E axis. See https://github.com/MarlinFirmware/Marlin/pull/24797#issuecomment-1333268311.

`struct DistinctAxes` is identical to `struct { T LOGICAL_AXIS_ARGS(); }` from `template<typename T> struct XYZEval` for its first `NUM_AXES` elements. This anonymous member is used for initialisation of `xyze_float_t` and it should initialise the same way as `DistinctAxes`.